File: src/test/java/com/ds/inventory/PartServiceTests.java
package com.ds.inventory;

import java.io.*;
import java.util.*;
import java.util.stream.*;

import org.junit.jupiter.api.*;
import com.marklogic.client.*;
import com.ds.TestBase;
import com.ds.inventory.PartService;


public class PartServiceTests extends TestBase {
  private static DatabaseClient dbClient;
  private static PartService partService;

  @BeforeAll
  public static void setUp() {
    dbClient = getClient();
    partService = PartService.on(dbClient);
  }
  
  @AfterAll
  public static void tearDown() {
    dbClient.release();
  }

  @Test
  void testListParts() throws IOException {
    // parameters
    int pageLength = 10;
    List<String> options = Arrays.asList("the", "quick", "brown", "fox");
    StringReader doc = new StringReader("<test><node>1</node></test>");

    Stream<String> retVal = partService.listParts(pageLength, options.stream(), doc);
    
    List<String> results = retVal.collect(Collectors.toList());
    results.forEach(System.out::println);
  }
}

File: dotnet/MarkLogic.Client.Tests/PartServiceTests.cs
using System.IO;
using Xunit;
using Xunit.Abstractions;

namespace MarkLogic.Client.Tests
{
    public class PartServiceTests : IClassFixture<DatabaseClientFixture>
    {
        private readonly DatabaseClientFixture _dbClientFixture;
        private readonly ITestOutputHelper _output;

        public PartServiceTests(DatabaseClientFixture dbClientFixture, ITestOutputHelper output)
        {
            _dbClientFixture = dbClientFixture;
            _output = output;
        }

        private IDatabaseClient DbClient => _dbClientFixture.DbClient;

        [Fact]
        public async void TestListParts()
        {
            var pageLength = 10;
            var options = new string[] { "the", "quick", "brown", "fox" };
            var doc = new StringReader("<test><node>1</node></test>");

            var ps = PartService.Create(DbClient); 
            var results = await ps.listParts(pageLength, options, doc);

            foreach(var result in results)
            {
                _output.WriteLine(result);
            }
        }
    }
}
